.github/workflows/dev-setup.yml

```diff
@@ -0,0 +1,50 @@
+name: Dev environment setup
+
+on:
+  push:
+    paths:
+      - 'dev/dev-env-setup.sh'
+      - 'dev/test-dev-env-setup.sh'
+      - '.github/workflows/dev-setup.yml'
+  schedule:
+    - cron: "42 7 * * 0"
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: >
+          [Optional] Repository name with owner. For example, mlflow/mlflow.
+           Defaults to the repository that triggered a workflow.
+        required: false
+        default: ""
+      ref:
+        description: >
+          [Optional] The branch, tag or SHA to checkout. When checking out the repository that
+           triggered a workflow, this defaults to the reference or SHA for that event. Otherwise,
+           uses the default branch.
+        required: false
+        default: ""
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+jobs:
+  linux-env-setup:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.ref }}
+      - uses: ./.github/actions/setup-pyenv
```

Comment:
```suggestion
```

Can we remove this line and test pyenv installation?

Reply:
definitely! Removing and validating CI build.